enhancements/node-tuning/mixed-cpu-node-plugin.md

```diff
@@ -0,0 +1,417 @@
+A way of mitigating that is to use workload partitioning
+to ensure the platform housekeeping processes don't run on the shared cpus.
+
+* There is no risk that workload’s tasks run on dedicated OCP’s housekeeping reserved cpus,
+because only the shared cpus exposed via cgroups to the workload’s process.
+In other words, platform's housekeeping processes can run on shared cpus dedicated to workload’s light-weight tasks,
+but not the other way around.
+
+* mixed-cpus-node-plugin is NRI-based plugin.
+Bugs in the plugin or NRI framework might affect the containers at various lifecycle stages, e.g.,
+CreateContainer, UpdateContainer, etc.
+For mitigating such cases, it is possible to deactivate the node-plugin, and in severe cases it is possible to disable
+NRI completely at runtime-level.
+
+* From a performance perspective, the runtime communicates with the node-plugin via gRPC calls.
+Therefore, for every container creation, there's another gRPC call between the runtime and the node-plugin.
+We can't avoid this call even for containers that don't ask for shared CPUs, since runtime doesn't have any filtering
+mechanism, it sends all the containers through the NRI framework to node-plugin, and the filtering is done at the
+plugin side.
+
+### Drawbacks
+
+The way of how user should ask for shared CPUs in the pod spec is through device plugin.
+While this approach fits in the model of how the workload requires resources,
+better integration would require more invasive changes which are out of scope now.
+    
+## Design Details
+N/A
+
+### Test Plan
+
+#### Unit testing
+We will have unit-testing for the mixed-cpu-node-plugin to guarantee the basic functionality.
+
+#### E2E testing on mixed-cpu-node-plugin
+The behavior of the mixed-cpu-node-plugin will be validated in isolation (without NTO) by E2E tests
+
+#### E2E testing on NTO
+Validating a deployment process on NTO, verifying PerformanceProfile configuration and checking 
+the functionality of mixed-cpu-node-plugin (running the e2e tests from mixed-cpu-node-plugin) 
+
+### Graduation Criteria
+N/A
+
+#### Dev Preview -> Tech Preview
+N/A
+
+#### Tech Preview -> GA
+N/A
+
+#### Removing a deprecated feature
+N/A
+
+### Upgrade / Downgrade Strategy
+N/A
+
+### Version Skew Strategy
+N/A
+
+### Operational Aspects of API Extensions
+N/A
+
+#### Failure Modes
+N/A
+
+#### Support Procedures
+N/A
+
+## Implementation History
+N/A
+
+## Alternatives
+* Implement directly in CRI-O.
+The user specifies an annotation for shared-cpus and runs with high-performance runtime class under its workload.
+Instead of calling NRI hook, CRI-O executes all the logic needed for the shared CPUs injection.
+
+Pros:
+1. Simplifies the solution by avoiding the need of running with DaemonSet 
+2. No need to handle image artifacts/delivery
+3. CRI-O always starts before all the pods, so no racing issue on startup.
+
+Cons:
+1. Vendor lock-in solution. cannot be used with other runtimes.
```

Comment:
I think this isn't a valid con since we are already locked in mgmt core pinning.

Reply:
@mrunalp I know about at least one stakeholder that would prefer to avoid vendor lock-in as much as possible.
We can discuss about it further in internal channels if you like.